cns/restserver/ipam.go

```diff
@@ -92,26 +92,38 @@ func (service *HTTPRestService) releaseIPConfigHandler(w http.ResponseWriter, r
 	return
 }
 
+// MarkIPsAsPending will mark IPs to pending release state.
 func (service *HTTPRestService) MarkIPsAsPending(numberToMark int) (map[string]cns.IPConfigurationStatus, error) {
 	pendingReleaseIPs := make(map[string]cns.IPConfigurationStatus)
-	markedIPCount := 0
+	// Ensure PendingProgramming IPs will be release before Available ones.
+	ipStateTypes := [2]string{cns.PendingProgramming, cns.Available}
 
 	service.Lock()
 	defer service.Unlock()
-	for uuid, _ := range service.PodIPConfigState {
+	for _, ipStateType := range ipStateTypes {
+		pendingReleaseIPs := service.markSepcificIPTypeAsPending(numberToMark, len(pendingReleaseIPs), ipStateType, pendingReleaseIPs)
```

Comment:
there's a possible bug here. the `pendingReleaseIPs` variable is redeclared in this scope, so what executes on every iteration is equivalent to
```go
service.markSpecificIPTypeAsPending(numberToMark, 0, ipStateType, /*empty map*/)
```

Reply:
Modified to avoid redeclaration.